### PR TITLE
more failures

### PR DESCRIPTION
**Pull request by anxcon**

### CODE REVIEW

File: Failure Modules/TestFlightFailure_SolarTracking.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            base.DoRepair();
+            if (base.part != null && base.part.partInfo != null)
+            {
+                Part pf = base.part.partInfo.partPrefab;
+                if (pf != null)
+                {
+                    ModuleDeployableSolarPanel pm = (ModuleDeployableSolarPanel)pf.Modules.OfType<ModuleDeployableSolarPanel>();
+                    if (pm != null)
+                    {
+                        base.module.trackingSpeed = pm.trackingSpeed;
```
Comment (jwvanderbeck) [inline Failure Modules/TestFlightFailure_SolarTracking.cs]: Wouldn't it be preferable to cache the previous tracking speed in DoFailure and then simply restore it here?

Comment (anxcon) [inline Failure Modules/TestFlightFailure_SolarTracking.cs]: any reasoning behind that thinking? before i used to save the value, but since the field trackingSpeed isnt persistent, and nothing to tweak with it, figured better to use the prefab. once failed state of failures is fixed to save, id have to save the cached value as well. 1) adding more data to the save 2) messing up any changes from configs that change the part. if speed is 5, fail saves to 0, new mod added changes to 10, repair would put back 5, until restart where 10 fills in anyways.

feel free to say i overthink it, just what felt right, i'm not attached to either method, so feel free to discuss :)

Comment (jwvanderbeck) [inline Failure Modules/TestFlightFailure_SolarTracking.cs]: That scenario would only happen in the very specific case of

1. Part fails
2. Game is saved
3. New mod is installed changing default value
4. Game with failed part is loaded

While it certainly can happen, it is definitely in the minority.  On the other hand what if the player has another mod installed that is changing tracking speed at runtime based on something, or even what if you or someone else makes another failure mod that slows tracking speed rather than stops it.  In both cases, if you set the tracking speed back to the prefab value, that would be the wrong value.  But if you cache it, you will always know you are setting it back to what it was immediately prior to failure.

In other words caching it is going to give you a "fixed" value more often than not, and as such it is the general rule I use when modifying values at runtime in failures.

Comment (anxcon) [inline Failure Modules/TestFlightFailure_SolarTracking.cs]: hmm fair points, guess il change and cache it :)
since github added my latest work of art to the wrong branch anyways lol
